src/Microsoft.AspNetCore.StaticFiles/DirectoryBrowserMiddleware.cs

```diff
@@ -46,7 +53,7 @@ public DirectoryBrowserMiddleware(RequestDelegate next, IHostingEnvironment host
 
             if (options.Value.Formatter == null)
             {
-                throw new ArgumentException(Resources.Args_NoFormatter);
+                options.Value.Formatter = new HtmlDirectoryFormatter(encoder);
```

Comment:
Okay you asked for it, now I'm going to do some larger surgery then if I'm going to be touching more than a line or two :)